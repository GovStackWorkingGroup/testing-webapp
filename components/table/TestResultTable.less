@import '../../styles/Variables.less';
@import '../../styles/Extends.less';

.test-table-container {
  display: flex;
  flex-direction: column;
  padding-top: 16px;
  padding-bottom: 16px;
  gap: 16px;
  height: 100%;
}

.test-table-last-update,
.test-table-result {
  display: flex;
  flex-direction: row;
  gap: 5px;
  font-size: 12px;
  color: @text-secondary;
}

.test-table-header:extend(.table-header-text) {
  display: grid;
  grid-template-columns: 1fr 1fr 2fr;
  grid-template-rows: @table-row;
  padding: 0 24px;
  border: 1px solid @stroke;
  border-radius: 8px 8px 0px 0px;
  align-items: center;
  background-color: #fafbff;
}

.test-table-row {
  height: @table-row;
  display: grid;
  grid-template-columns: 1fr 1fr 2fr;
  padding: 0 24px;
  border-bottom: 1px solid @stroke;
  border-right: 1px solid @stroke;
  border-left: 1px solid @stroke;
  align-items: center;
  color: @text-secondary;
}

.test-table-row:hover {
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.25);
}

.test-table-row-status,
.test-table-row-category {
  width: max-content;
  height: 22px;
  border-radius: 4px;
  line-height: 22px;
  padding: 4px 8px;
}

.test-table-row-status {
  font-size: 12px;
}

.test-table-row-category {
  background-color: @background-category;
  color: @accents-and-btn;
  font-size: 10px;
}

.test-table-row-endpoint {
  overflow: hidden;
}

.test-table-row-endpoint > p {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.status-failed {
  background-color: @fail-background;
  color: @fail;
}

.status-passed {
  background-color: @success-backround;
  color: @success;
}

.row-endpoint-tooltip {
  overflow-wrap: anywhere;
  background-color: @background-page;
  color: @text-secondary;
  font-size: 12px;
  padding: 10px 16px;
  box-shadow: 0px 1px 4px @blue-20;
  border: 1px solid @blue-20;
}
