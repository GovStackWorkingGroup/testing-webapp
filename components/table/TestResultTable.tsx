import { useRouter } from 'next/router';
import React, { useCallback } from 'react';
import { useIntl } from 'react-intl';
import { BuildingBlockTestSummary } from '../../service/types';
import TableErrorHandling from './TableErrorHandling';
import TestResultTableHeader from './TestResultTableHeader';
import TestResultTableRow from './TestResultTableRow';

type Props = {
  bbSummary: BuildingBlockTestSummary | undefined
}

const TestResultTable = ({ bbSummary }: Props) => {
  const router = useRouter();
  const { formatMessage } = useIntl();

  const format = useCallback(
    (id: string) => formatMessage({ id }),
    [formatMessage]
  );

  const { locale } = router;

  return (
    <div className='test-table-container'>
      <div className='test-table-last-update' data-testid='test-table-last-update'>
        <p>{format('test_table.last_update.label')}</p>
        <p>{bbSummary ? new Date(bbSummary?.compatibilities.saveTime).toLocaleDateString(locale) : '-'}</p>
      </div>
      <div className='test-table-result' data-testid='test-table-result'>
        {!bbSummary?.count ? <p>{'-'}</p> : <p>{bbSummary.count}</p>}
        <p>
          {bbSummary?.count === 1
            ? format('table.result.label')
            : format('table.result.plural.label')}
        </p>
      </div>
      <div>
        <TestResultTableHeader />
        {!bbSummary?.data ? <TableErrorHandling /> : (
          bbSummary.data.map((bbTest, idx )=> <TestResultTableRow bbTest={bbTest} key={`bbTest-${idx}`} />)
        )}

      </div>
    </div>
  );
};

export default TestResultTable;
