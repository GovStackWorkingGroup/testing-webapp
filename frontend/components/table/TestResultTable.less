@import '../../styles/Variables.less';
@import '../../styles/Extends.less';

.test-table-container {
  display: flex;
  flex-direction: column;
  padding-top: 16px;
  padding-bottom: 16px;
  gap: 16px;
  height: calc(100vh - 320px);
}

.test-table-last-update,
.test-table-result {
  display: flex;
  flex-direction: row;
  gap: 5px;
  font-size: 12px;
  color: @text-secondary;
}

.test-table-table {
  max-height: calc(100% - 55px);
  height: fit-content;
  border: 1px solid @stroke;
  border-radius: 8px 8px 0px 0px;
}

.test-table-header:extend(.table-header-text) {
  display: grid;
  grid-template-columns: 1fr 1fr 2fr;
  grid-template-rows: @table-row;
  padding: 0 24px;
  border-bottom: 1px solid @stroke;
  border-radius: 8px 8px 0px 0px;
  align-items: center;
  background-color: #fafbff;
}

.test-table-header > div {
  display: flex;
  gap: 11px;
  align-items: center;
  cursor: pointer;
}

.test-table-row-container {
  overflow-y: scroll;
  height: calc(100% - 50px);
}

.test-table-row {
  height: @table-row;
  display: grid;
  grid-template-columns: 1fr 1fr 2fr;
  padding: 0 24px;
  border-bottom: 1px solid @stroke;
  align-items: center;
  color: @text-secondary;
}

.test-table-row:hover {
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.25);
}

.test-table-row-status,
.test-table-row-category {
  width: max-content;
  height: 22px;
  border-radius: 4px;
  line-height: 22px;
  padding: 4px 8px;
}

.test-table-row-status {
  font-size: 12px;
}

.test-table-row-category {
  background-color: @background-category;
  color: @accents-and-btn;
  font-size: 10px;
}

.test-table-row-endpoint {
  overflow: hidden;
}

.test-table-row-endpoint > p {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.status-failed {
  background-color: @fail-background;
  color: @fail;
}

.status-passed {
  background-color: @success-background;
  color: @success;
}

.row-endpoint-tooltip {
  overflow-wrap: anywhere;
  background-color: @background-page;
  color: @text-secondary;
  font-size: 12px;
  padding: 10px 16px;
  box-shadow: 0px 1px 4px @blue-20;
  border: 1px solid @blue-20;
}
