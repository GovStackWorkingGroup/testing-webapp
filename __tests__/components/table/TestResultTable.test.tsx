import { render } from '../../test-utils/test-utils';
import TestResultTable from '../../../components/table/TestResultTable';

describe('Unit tests for TestResultTable component:', () => {
  it('should match snapshot', () => {
    const { container } = render(<TestResultTable />);

    expect(container).toMatchSnapshot();
  });
});
