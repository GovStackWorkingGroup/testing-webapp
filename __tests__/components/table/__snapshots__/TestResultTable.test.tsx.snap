// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for TestResultTable component: should match snapshot 1`] = `
<div>
  <div
    class="test-table-container"
  >
    <div
      class="test-table-last-update"
      data-testid="test-table-last-update"
    >
      <p>
        Last update:
      </p>
      <p>
        8/1/2022
      </p>
    </div>
    <div
      class="test-table-result"
      data-testid="test-table-result"
    >
      <p>
        1
      </p>
      <p>
         result
      </p>
    </div>
    <div>
      <div
        class="test-table-header"
      >
        <div
          data-testid="header-status-label"
        >
          <p>
            Status
          </p>
        </div>
        <div
          data-testid="header-category-label"
        >
          <p>
            Category
          </p>
        </div>
        <div
          data-testid="header-name-label"
        >
          <p>
            Name
          </p>
        </div>
      </div>
      <div
        class="test-table-row"
      >
        <div
          class="test-table-row-status status-passed"
          data-testid="bb-test-passed"
        >
          <p>
            Passed
          </p>
        </div>
        <div
          class="test-table-row-category"
          data-testid="bb-test-category"
        >
          GET
        </div>
        <div
          data-testid="bb-test-endpoint"
        >
          <p>
            /data/{someparameter}/method
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
